In [29]:
import pandas as pd
import plotly.express as px

In [30]:
df = pd.read_csv('Project_dataset_1.csv')

for index in range(1,len(df.columns)):
    if index < 5 and index > 1:
        mode = ' road'
    elif index > 4:
        mode = ' PT'
    else:
        mode = ''
    df = df.rename({ df.columns[index] : df.iloc[0, index]+mode}, axis = 1)
df = df.drop(df.index[0:2], axis = 0)
df = df.rename({'Unnamed: 0' : 'Demographic group'}, axis = 1)
df = df.rename({'Sex' : 'year'}, axis = 1)
df = df.drop(137, axis = 0)
df



Demographic group  year Total male and female road  \
2                       Total persons  2018                       82.9   
3                       Total persons  2019                       81.9   
4                       Total persons  2020                       73.8   
5                       Total persons  2021                       78.1   
6                       Total persons  2022                       79.7   
..                                ...   ...                        ...   
132  Participation: unemployed, other  2018                       73.2   
133  Participation: unemployed, other  2019                       71.0   
134  Participation: unemployed, other  2020                       62.1   
135  Participation: unemployed, other  2021                       67.8   
136  Participation: unemployed, other  2022                       70.1   

    Male road Female road Total male and female PT Male PT Female PT  
2        83.0        82.9                      8.6     8.0       9.2  
3        82.6        81.3                      8.6     7.9       9.2  
4        74.4        73.2                      3.9     3.6       4.2  
5        77.9        78.4                      4.2     3.8       4.7  
6        79.8        79.7                      6.1     5.6       6.5  
..        ...         ...                      ...     ...       ...  
132      73.4        73.1                      5.4     6.3       5.1  
133      72.2        70.6                      5.0     6.4       4.5  
134      64.4        61.4                      2.1       .       1.9  
135      67.7        67.8                      2.9       .       2.8  
136      70.0        70.1                      3.8     5.4       3.3  

[135 rows x 8 columns]

In [31]:
for row in range(df.shape[0]):
    for column in range(2,df.shape[1]):
        df.iloc[row,column] = pd.to_numeric(df.iloc[row,column], errors='coerce')
df

Demographic group  year Total male and female road  \
2                       Total persons  2018                       82.9   
3                       Total persons  2019                       81.9   
4                       Total persons  2020                       73.8   
5                       Total persons  2021                       78.1   
6                       Total persons  2022                       79.7   
..                                ...   ...                        ...   
132  Participation: unemployed, other  2018                       73.2   
133  Participation: unemployed, other  2019                       71.0   
134  Participation: unemployed, other  2020                       62.1   
135  Participation: unemployed, other  2021                       67.8   
136  Participation: unemployed, other  2022                       70.1   

    Male road Female road Total male and female PT Male PT Female PT  
2        83.0        82.9                      8.6     8.0       9.2  
3        82.6        81.3                      8.6     7.9       9.2  
4        74.4        73.2                      3.9     3.6       4.2  
5        77.9        78.4                      4.2     3.8       4.7  
6        79.8        79.7                      6.1     5.6       6.5  
..        ...         ...                      ...     ...       ...  
132      73.4        73.1                      5.4     6.3       5.1  
133      72.2        70.6                      5.0     6.4       4.5  
134      64.4        61.4                      2.1     NaN       1.9  
135      67.7        67.8                      2.9     NaN       2.8  
136      70.0        70.1                      3.8     5.4       3.3  

[135 rows x 8 columns]

In [32]:
df['gap with mean']=0
for i in range(2, df.shape[0]+2):
    ref_year = df.loc[i, 'year']
    ref_value = df[(df['Demographic group'] == 'Total persons') & (df['year'] == ref_year)].iloc[0, 5]
    df.loc[i, 'gap with mean'] = df.loc[i, 'Total male and female PT'] - ref_value

df


Demographic group  year Total male and female road  \
2                       Total persons  2018                       82.9   
3                       Total persons  2019                       81.9   
4                       Total persons  2020                       73.8   
5                       Total persons  2021                       78.1   
6                       Total persons  2022                       79.7   
..                                ...   ...                        ...   
132  Participation: unemployed, other  2018                       73.2   
133  Participation: unemployed, other  2019                       71.0   
134  Participation: unemployed, other  2020                       62.1   
135  Participation: unemployed, other  2021                       67.8   
136  Participation: unemployed, other  2022                       70.1   

    Male road Female road Total male and female PT Male PT Female PT  \
2        83.0        82.9                      8.6     8.0       9.2   
3        82.6        81.3                      8.6     7.9       9.2   
4        74.4        73.2                      3.9     3.6       4.2   
5        77.9        78.4                      4.2     3.8       4.7   
6        79.8        79.7                      6.1     5.6       6.5   
..        ...         ...                      ...     ...       ...   
132      73.4        73.1                      5.4     6.3       5.1   
133      72.2        70.6                      5.0     6.4       4.5   
134      64.4        61.4                      2.1     NaN       1.9   
135      67.7        67.8                      2.9     NaN       2.8   
136      70.0        70.1                      3.8     5.4       3.3   

     gap with mean  
2              0.0  
3              0.0  
4              0.0  
5              0.0  
6              0.0  
..             ...  
132           -3.2  
133           -3.6  
134           -1.8  
135           -1.3  
136           -2.3  

[135 rows x 9 columns]

In [33]:
df['comparison to 2018 level']=0
for i in range(2, df.shape[0]+2):
    ref_group = df.loc[i, 'Demographic group']
    ref_value = df[(df['Demographic group'] == ref_group) & (df['year'] == '2018')].iloc[0, 5]
    comparison_value = df.loc[i, 'Total male and female PT']
    df.loc[i, 'comparison to 2018 level'] = comparison_value - ref_value

df

Demographic group  year Total male and female road  \
2                       Total persons  2018                       82.9   
3                       Total persons  2019                       81.9   
4                       Total persons  2020                       73.8   
5                       Total persons  2021                       78.1   
6                       Total persons  2022                       79.7   
..                                ...   ...                        ...   
132  Participation: unemployed, other  2018                       73.2   
133  Participation: unemployed, other  2019                       71.0   
134  Participation: unemployed, other  2020                       62.1   
135  Participation: unemployed, other  2021                       67.8   
136  Participation: unemployed, other  2022                       70.1   

    Male road Female road Total male and female PT Male PT Female PT  \
2        83.0        82.9                      8.6     8.0       9.2   
3        82.6        81.3                      8.6     7.9       9.2   
4        74.4        73.2                      3.9     3.6       4.2   
5        77.9        78.4                      4.2     3.8       4.7   
6        79.8        79.7                      6.1     5.6       6.5   
..        ...         ...                      ...     ...       ...   
132      73.4        73.1                      5.4     6.3       5.1   
133      72.2        70.6                      5.0     6.4       4.5   
134      64.4        61.4                      2.1     NaN       1.9   
135      67.7        67.8                      2.9     NaN       2.8   
136      70.0        70.1                      3.8     5.4       3.3   

     gap with mean  comparison to 2018 level  
2              0.0                       0.0  
3              0.0                       0.0  
4              0.0                      -4.7  
5              0.0                      -4.4  
6              0.0                      -2.5  
..             ...                       ...  
132           -3.2                       0.0  
133           -3.6                      -0.4  
134           -1.8                      -3.3  
135           -1.3                      -2.5  
136           -2.3                      -1.6  

[135 rows x 10 columns]

In [34]:
df['comparison to 2018 level (%)']=0
for i in range(2, df.shape[0]+2):
    ref_group = df.loc[i, 'Demographic group']
    ref_value = df[(df['Demographic group'] == ref_group) & (df['year'] == '2018')].iloc[0, 5]
    comparison_value = df.loc[i, 'Total male and female PT']
    df.loc[i, 'comparison to 2018 level (%)'] = ((comparison_value - ref_value)/ref_value)*100

df

Demographic group  year Total male and female road  \
2                       Total persons  2018                       82.9   
3                       Total persons  2019                       81.9   
4                       Total persons  2020                       73.8   
5                       Total persons  2021                       78.1   
6                       Total persons  2022                       79.7   
..                                ...   ...                        ...   
132  Participation: unemployed, other  2018                       73.2   
133  Participation: unemployed, other  2019                       71.0   
134  Participation: unemployed, other  2020                       62.1   
135  Participation: unemployed, other  2021                       67.8   
136  Participation: unemployed, other  2022                       70.1   

    Male road Female road Total male and female PT Male PT Female PT  \
2        83.0        82.9                      8.6     8.0       9.2   
3        82.6        81.3                      8.6     7.9       9.2   
4        74.4        73.2                      3.9     3.6       4.2   
5        77.9        78.4                      4.2     3.8       4.7   
6        79.8        79.7                      6.1     5.6       6.5   
..        ...         ...                      ...     ...       ...   
132      73.4        73.1                      5.4     6.3       5.1   
133      72.2        70.6                      5.0     6.4       4.5   
134      64.4        61.4                      2.1     NaN       1.9   
135      67.7        67.8                      2.9     NaN       2.8   
136      70.0        70.1                      3.8     5.4       3.3   

     gap with mean  comparison to 2018 level  comparison to 2018 level (%)  
2              0.0                       0.0                      0.000000  
3              0.0                       0.0                      0.000000  
4              0.0                      -4.7                    -54.651163  
5              0.0                      -4.4                    -51.162791  
6              0.0                      -2.5                    -29.069767  
..             ...                       ...                           ...  
132           -3.2                       0.0                      0.000000  
133           -3.6                      -0.4                     -7.407407  
134           -1.8                      -3.3                    -61.111111  
135           -1.3                      -2.5                    -46.296296  
136           -2.3                      -1.6                    -29.629630  

[135 rows x 11 columns]

In [35]:
mask = df['Demographic group'].str.startswith('Total')
df_total = df[mask]

In [36]:
mask = df['Demographic group'].str.startswith('Age')
df_age = df[mask]
df_age2 = pd.concat([df_total, df_age])

In [37]:
px.line(data_frame=df_age2, x="year", y="Total male and female PT", color='Demographic group')

#### Observations :
The drop of traffic during covid is more sharp for younger categories (age < 35 years) compared to the other age groups. Maybe this can be explained by the closing of schools and universities. Maybe younger workers also tend to use remote working more than older ones.

In [38]:
px.line(data_frame=df_age, x="year", y="gap with mean", color='Demographic group')

#### Observations :
What is interesting to note is that for people 35 years old and older, the gap with the mean is quite constant during the pandemic, and is even reduced for some categories. Post-covid gaps are generally less important but still negative. Thus, we can imagine that people aged 35 and more were less impacted by the pandemic than other categories. This can maybe be explained by the fact that a lot of younger persons, who make up the bulk of public transport traffic, stopped using public transport because of schools and universities closing, whereas certain necessary workplaces were still open.
 

In [39]:
px.bar(data_frame = df_age2[(df_age2['year'] == '2022')], x='Demographic group', y='comparison to 2018 level', color='Demographic group')

#### Observations :
In terms of raw participation, the group with the sharpest drop is the 18 to 24 years age group. As mentionned before, this can maybe be explained by the fact during the pandemic, many universities were closed, so universities implemented more online courses, which may have reduced the transportation needs of students after the pandemic. Also, maybe people preferred using individual transport means (bikes, cars) during the pandemic to avoid restrictions and kept this habit afterwards.
The second most impacted group are people from 25 to 34 years old. Maybe this can be explained by an increase in remote work after the pandemic
However, we have to keep in mind that the variation is bigger in these groups because they are already the groups with the highest participation

In [40]:
px.bar(data_frame = df_age2[(df_age2['year'] == '2022')], x='Demographic group', y='comparison to 2018 level (%)', color='Demographic group')

In terms or percentages, the average variation (variation for the total) is - 29%.
Even if the variation was higher in terms of raw numbers for the younger age groups, proportionnaly the drop was bigger for age groups over 35 years old.
One explanation could be that, since these groups were already using public transports less than other groups, people from these older age groups were more likely to stop taking public transport, or at least decrease significantly their use of public transport. Furthermore, maybe these groups have more alternatives to public transport than younger age groups, for example access to a car.